program.c

```diff
@@ -80,6 +81,90 @@ static int load_erase_tag(xmlNode *node, bool is_nand)
 	return 0;
 }
 
+static struct program *program_load_sparse(struct program *program, int fd)
+{
+	struct program *program_sparse = NULL;
+	struct program *programes_sparse = NULL;
+	struct program *programes_sparse_last = NULL;
+	char tmp[PATH_MAX];
+
+	sparse_header_t sparse_header;
+	unsigned start_sector, chunk_size, chunk_type, chunk_data;
+
+	if (sparse_header_parse(fd, &sparse_header)) {
+		/* ignore false sparse images */
+		if (program->sector_size * program->num_sectors == lseek(fd, 0, SEEK_END)) {
+			program_sparse = calloc(1, sizeof(struct program));
+			memcpy(program_sparse, program, sizeof(struct program));
+			program_sparse->sparse = false;
+			program_sparse->next = NULL;
+			return program_sparse;
+		}
+
+		fprintf(stderr, "[PROGRAM] Unable to parse sparse header at %s...failed\n",
+				program->filename);
+		return NULL;
+	}
+
+	for (uint32_t i = 0; i < sparse_header.total_chunks; ++i) {
+
```

Comment:
As you're in this function fixing anyways, could you please drop this blank line?

Reply:
removed